### PR TITLE
Add base64 built-in functions.

### LINKED ISSUE #125: Built-in functions for base64 encoding/decoding

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
My current implementation also adds functions to convert between strings and UTF-8 byte arrays. To be more explicit, a conversion to a UTF-8 byte array is needed before converting these bytes to base64, and vice versa.

This PR will close #125.

### CODE REVIEW

File: src/Scriban/Functions/StringFunctions.cs [hunk 1]
```diff
@@ -971,10 +971,66 @@ public static string PadLeft(string text, int width)
...
         {
             return (text ?? string.Empty).PadRight(width);
         }
+
+        /// <summary>
+        ///  Encodes a string into an array of UTF-8-encoded bytes.
+        /// </summary>
+        /// <param name="text">The string to encode</param>
+        /// <returns>The encoded bytes</returns>
+        public static byte[] Utf8Encode(string text)
```
Comment (xoofx) [inline src/Scriban/Functions/StringFunctions.cs]: this will require to regenerate the builtin delegates (run `Scriban.CodeGen` in the project)

Comment (teo-tsirpanis) [reply src/Scriban/Functions/StringFunctions.cs]: It doesn't. Unless something is wrong... Wait a minute.

image

**Comment by teo-tsirpanis** [comment]
Are you OK with the API? Another idea is to _assume_ the encoding to be UTF-8 and don't work with byte arrays at all.

I have no personal preference between either, actually.

File: src/Scriban/Functions/StringFunctions.cs (hunk 1 above)
Comment (teo-tsirpanis) [reply src/Scriban/Functions/StringFunctions.cs]: Is it normal? Should new code be generated? Did I break anything?

Comment (xoofx) [reply src/Scriban/Functions/StringFunctions.cs]: I don't know,, maybe the problem is not handling this kind of parameter types, it is the first time that there is a `byte[]` so maybe there was an exception when you ran it?

Comment (teo-tsirpanis) [reply src/Scriban/Functions/StringFunctions.cs]: No, there was no exception. The process returned 0.

**Comment by xoofx** [comment]
> Are you OK with the API? Another idea is to _assume_ the encoding to be UTF-8 and don't work with byte arrays at all.

The problem is that byte arrays are completely new, and well, overall the API has been designed to manipulate strings... byte[] arrays are not really printable... so maybe you could remove it, because it is barely usable without exposing an object with an array... also, the type in itself (byte[]) might not be handled well across the runtime.

File: src/Scriban/Functions/StringFunctions.cs (hunk 1 above)
Comment (xoofx) [reply src/Scriban/Functions/StringFunctions.cs]: I don't know, you will have to debug, maybe something different that doesn't work well on your setup, or a bug in the program itself

**Comment by teo-tsirpanis** [comment]
> it is barely usable

I was aware, but I also saw the array functions operating on `IEnumerable` and other collection interfaces. I will make conversions from strings to strings directly.

**Comment by xoofx** [comment]
Thanks!

**Comment by teo-tsirpanis** [comment]
The code generator has a bug I am investigating right now.

Can you merge this PR so that my next one builds on top of this one? Not all builtins are in `CustomFunction.Generated.cs` after all.

**Comment by teo-tsirpanis** [comment]
> Can you merge this PR

Never mind, you already did. 😂

**Pull request merged**